modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/diagnostic/pagelocktracker/dumpprocessors/ToStringDumpHelperTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.diagnostic.pagelocktracker.dumpprocessors;
+
+import java.lang.reflect.Field;
+import org.apache.ignite.internal.processors.cache.persistence.diagnostic.pagelocktracker.PageLockDump;
+import org.apache.ignite.internal.processors.cache.persistence.diagnostic.pagelocktracker.SharedPageLockTracker;
+import org.apache.ignite.internal.processors.cache.persistence.diagnostic.pagelocktracker.SharedPageLockTrackerDump;
+import org.apache.ignite.internal.processors.cache.persistence.diagnostic.pagelocktracker.ThreadPageLockState;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link ToStringDumpHelper}.
+ */
+public class ToStringDumpHelperTest extends GridCommonAbstractTest {
+
+    /** */
+    @Test
+    public void toStringSharedPageLockTrackerTest() throws Exception {
+        SharedPageLockTracker pageLockTracker = new SharedPageLockTracker();
+
+        pageLockTracker.onReadLock(1, 2, 3, 4);
+
+        pageLockTracker.onReadUnlock(1, 2, 3, 4);
+
+        Thread asyncLockUnlock = new Thread(() -> {
+            pageLockTracker.onReadLock(4, 32, 1, 64);
+        }, "async-lock-unlock");
+        asyncLockUnlock.start();
+        asyncLockUnlock.join();
+        long threadIdInLog = asyncLockUnlock.getId();
+
+        SharedPageLockTrackerDump pageLockDump = pageLockTracker.dump();
+
+        // hack to have same timestamp in test
+        Field timeField = PageLockDump.class.getDeclaredField("time");
+        timeField.setAccessible(true);
+
+        for (ThreadPageLockState state : pageLockDump.threadPageLockStates) {
+            timeField.setLong(state.pageLockDump, 1596173397167L);
+        }
+
+        assertNotNull(pageLockDump);
+
+        String dumpStr = ToStringDumpHelper.toStringDump(pageLockDump);
+
+        System.out.println("Dump saved:" + dumpStr);
```

Comment:
There's no point in that, string will be dumped if it doesn't match with the expected result.